app/Http/Controllers/BookingController.php

```diff
@@ -2,7 +2,87 @@
 
 namespace App\Http\Controllers;
 
+use App\Models\Booking;
+use App\Models\Room;
+use App\Services\JsonResponseService;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Log;
+
 class BookingController extends Controller
 {
-    //
+    private JsonResponseService $responseService;
+
+    public function __construct(JsonResponseService $responseService)
+    {
+        $this->responseService = $responseService;
+    }
+
+    public function create(Request $request)
+    {
+        $hidden = ['created_at', 'updated_at'];
+        $booking = new Booking();
+
+        $booking->customer = $request->customer;
+        $booking->guests = $request->guests;
+        $booking->start = $request->start;
+        $booking->end = $request->end;
+
+        if ($request->end < $request->start) {
+            return response()->json($this->responseService->getFormat(
+                'Start date must be before the end date'), 400);
+        }
```

Comment:
You should do this before you create the booking - remember as much as possible you want to try and get your methods to fail early